### PR TITLE
Add overwrite_conf option for metric agent

### PR DESCRIPTION
**Pull request by matrixik**
Change-Id: I268c1091824db41a5c5d2be3a160def9dd1364dc
Signed-off-by: Dobroslaw Zybort <dobroslaw.zybort@ts.fujitsu.com>

### CODE REVIEW

**Comment by adriancz** [comment]
Pleas add information to README

**Comment by matrixik** [comment]
Done

File: configure_metrics_agent.sh [hunk 1]
```diff
@@ -124,6 +175,29 @@ set_attributes() {
     inf "Set proper attributes successfully"
 }
 
+OVERWRITE_CONF=false
+
```
Comment (haruki-yamanashi) [inline configure_metrics_agent.sh]: Please remove this line.

File: configure_metrics_agent.sh [hunk 2]
```diff
@@ -10,18 +10,65 @@ BIN_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
...
+        return
+    fi
+}
+
+function run_monasca_setup() {
+    local all_args="$1"
+
+    # All this files will be uncoditionaly overwritten by monasca-setup
+    # so we are creating they backups if OVERWRITE_CONF is set to false
+    protect_overwrite "${MON_AGENT_DIR}/agent.yaml"
```
Comment (haruki-yamanashi) [inline configure_metrics_agent.sh]: I think it's better to define these 3 files as `list` and use for loop.

**Comment by haruki-yamanashi** [comment]
Please add/remove `function` for all method.

File: configure_metrics_agent.sh (hunk 1 above)
Comment (matrixik) [reply configure_metrics_agent.sh]: Done

File: configure_metrics_agent.sh (hunk 2 above)
Comment (matrixik) [reply configure_metrics_agent.sh]: Done

**Comment by matrixik** [comment]
All comments addressed.

**Review by adriancz — APPROVED** [review]

**Pull request merged**